src/tools/roc_conv/main.cpp

```diff
@@ -69,22 +70,26 @@ int main(int argc, char** argv) {
     pipeline::ConverterConfig config;
 
     if (args.frame_size_given) {
-        if (args.frame_size_arg <= 0) {
-            roc_log(LogError, "invalid --frame-size: should be > 0");
+        if (sizeof(args.frame_size_arg) <= 0) {
+            roc_log(LogError, "invalid --frame-size: frame size unspecified");
             return 1;
         }
-        config.internal_frame_size = (size_t)args.frame_size_arg;
+        roc::core::parse_duration(args.frame_size_arg, config.internal_frame_size);
+        if (config.internal_frame_size <= 0) {
+            roc_log(LogError, "invalid --frame-size: should be > 0");
+            return 1; 
+        }
     }
 
-    sndio::BackendDispatcher::instance().set_frame_size(config.internal_frame_size);
+    sndio::BackendDispatcher::instance().set_frame_size((unsigned long) config.internal_frame_size);
 
-    core::BufferPool<audio::sample_t> pool(allocator, config.internal_frame_size,
+    core::BufferPool<audio::sample_t> pool(allocator, (unsigned long) config.internal_frame_size,
                                            args.poisoning_flag);
 
     sndio::Config source_config;
     source_config.channels = config.input_channels;
     source_config.sample_rate = 0;
-    source_config.frame_size = config.internal_frame_size;
+    source_config.frame_size = (unsigned long) config.internal_frame_size;
```

Comment:
There are few more places like this where this conversion from frame length to frame size. I think its better if we get a function for this operation. How do we go about this?

Reply:
added ns_to_size().